pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -366,6 +368,17 @@ func (c *Cluster) checkHealth(ctx context.Context) error {
 		}
 	}
 
+	// failover mons running on host path to use persistent volumes if VolumeClaimTemplate is set
+	for _, monDeployment := range monDeployments.Items {
+		if _, ok := monDeployment.Labels["pvc_name"]; !ok {
+			if c.spec.Mon.VolumeClaimTemplate != nil {
+				monName := monDeployment.Labels[controller.DaemonIDLabel]
+				logger.Infof("failing over mon %q to use persistent volumes", monName)
+				c.failMon(len(c.ClusterInfo.Monitors), desiredMonCount, monName)
+			}
```

Comment:
Let's just failover one mon per health check, to follow the pattern of other failovers
```suggestion
				return nil
			}
```